src/components/Product/Product.js

```diff
@@ -0,0 +1,87 @@
+import { useState } from "react";
+import "./product.css";
+import PropTypes from "prop-types";
+import { useNavigate } from "react-router-dom";
+import Button from "../Button/Button";
+import Store from "../../contexts/Store";
+
+const Product = ({ data, extendedCard=false, disabledItem }) => {
+  const [disabled, setDisabled]  = useState(disabledItem);
+  const {
+    url,
+    name,
+    alt,
+    category,
+    price,
+    id,
+    short_description,
+    image,
+    stock,
+  } = data;
+  const navigate = useNavigate();
+  const goToProductDetail = () =>
+    navigate({
+      pathname: "/product",
+      search: "?productId=" + id,
+    });
+  return (
+    <Store.Consumer>
+      {({ cart, updateCart, getCartItem }) => {
+        const getDisabled = (cart, productId, numItems=0) => {
+          const cartItem = getCartItem(cart, productId);
+          if (cartItem.length > 0) {
+            setDisabled((prevDisabled) => cartItem[0].balance === numItems ? true : false);
```

Comment:
`cartItem[0].balance === numItems` is already a boolean expression, there is no need to use the ternary here, and the functional update is unnecessary as well:

```suggestion
            setDisabled(cartItem[0].balance === numItems);
```